openapi3/version.go

```diff
@@ -0,0 +1,42 @@
+package openapi3
+
+import (
+	"context"
+	"errors"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	versionRegex = regexp.MustCompile(`^3\.\d+\.\d+$`)
+
+	// ErrInvalidVersion is used when version is invalid (not 3.x.y)
+	ErrInvalidVersion = errors.New("must be 3.x.y")
+)
+
+// Version is specified by Version/Swagger standard version 3.
+// must be a sring
+type Version string
+
+// Validate returns an error if Schema does not comply with the Version spec.
+func (oai Version) Validate(ctx context.Context) error {
+	if vo := getValidationOptions(ctx); !vo.versionValidationDisabled {
+		if !versionRegex.MatchString(string(oai)) {
+			return ErrInvalidVersion
+		}
+	}
+	return nil
+}
+
+// MinorVersion returns minor version from string assuming 0 is the default
+func (oai Version) Minor() uint64 {
```

Comment:
`uint64` seems too big. Also why not return the two possible errors? (or edit func doc)

Reply:
I see you introduced this for use by the schema validation settings. Let's return a float then :)